### PR TITLE
BOM sketch

### LINKED ISSUE #40: BOM tag
In the 11th Jan 2023 meeting we have identified the need for a new BOM (bill of material) object that tells the verifier what a *complete* set of tags looks like.

The BOM object must list all the needed tags (CoMID, CoSWIDs, CoTS) using their IDs.

Additionally, the BOM will indicate the validity period of the aggregate.

A BOM object represents the signal for the verifier to *activate* the listed tags.  Data contained in a tag must not be used for appraisal until a BOM which activates that tag is received and successfully processed.  All the tags listed in the BOM must be activated in the same transaction, i.e., either all or none.

### PR DESCRIPTION
**Pull request by thomas-fossati**
(BOM lifecycle linking still missing)

Fixes #40 

Signed-off-by: Thomas Fossati <thomas.fossati@arm.com>
Signed-off-by: Yogesh Deshpande <yogesh.deshpande@arm.com>

### CODE REVIEW

File: cddl/concise-bom-tag.cddl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+concise-bom-tag = {
+  &(tag-identity: 0) => tag-identity-map
+  &(tags-list: 1) => [ + tag-identity-map ],
+  &(bom-validity: 2) => validity-map
+  ; TODO horizontal linking
```
Comment (nedmsmith) [inline cddl/concise-bom-tag.cddl:5]: should add extension point here

Comment (thomas-fossati) [reply cddl/concise-bom-tag.cddl]: done in [25a1738](https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/41/commits/25a1738c05d62395e46c5715d472644d21c5a104)

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1000,6 +1000,41 @@ measurements for the Target Environment.
 
 [^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/10
 
+# CoBOM {#sec-cobom}
+
+A BOM object represents the signal for the verifier to activate the listed
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1005]: Should expand the BOM acronym before using the acronym.
Also, wording should be more clear that the `tags-list` in the `concise-bom-tag` behaves like a signaling mechanism from a product vendor to a Verifier that _activates_ the tags in `tags-list` for use in an Evidence appraisal process.

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1000,6 +1000,41 @@ measurements for the Target Environment.
...
+tags. Data contained in a tag MUST NOT be used for appraisal until a BOM which
+activates that tag has been received and successfully processed. All the tags
+listed in the BOM must be activated in the same transaction, i.e., either all
+or none.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1009]: There should be normative language surrounding the scope of the activation. Possibly: All tags listed in `tags-list` MUST be activated or no tags are activated.

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1000,6 +1000,41 @@ measurements for the Target Environment.
...
+
+The CDDL specification for the `concise-bom-tag` map is as follows and this
+rule and its constraints MUST be followed when creating or validating a CoBOM
+tag:
+
+~~~ cddl
+{::include cddl/concise-bom-tag.cddl}
+~~~
+
+The following describes each member of the `concise-bom-tag` map.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1021]: Maybe the normative statements should follow (not precede) the description of the `concise-bom-tag` structure?

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1000,6 +1000,41 @@ measurements for the Target Environment.
 
 [^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/10
 
+# CoBOM {#sec-cobom}
+
+A BOM object represents the signal for the verifier to activate the listed
+tags. Data contained in a tag MUST NOT be used for appraisal until a BOM which
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1006]: Also we must say here that the tag_list comprises a list of tags (CoMID tags and/or CoSWID tags) that were successfully submitted to a verifier in past.

Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:1006]: not necessarily, they could be shipped contextually.  So the "present" also work :-)

File: draft-ietf-rats-corim.md (hunk 2 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md]: done in a5638b2

File: draft-ietf-rats-corim.md (hunk 3 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md]: done in a5638b2

File: draft-ietf-rats-corim.md (hunk 4 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:1021]: done in a5638b2

**Comment by thomas-fossati** [comment]
> I am okay with omitting horizontal linking as discussed in #40

Can I merge it then?  I'd need an approval from you lot :)

**Review by nedmsmith — APPROVED** [review]

**Pull request merged**